### PR TITLE
FWaaSv2.0 Policy CRUD

### PR DESCRIPTION
**Pull request by codemanufaktur**
follow up PR for  #1789 by @Elethiomel (I asked for the permission to get this done)
contains 2 cherry-picks from https://github.com/Daimler/gophercloud/commits/fwaas-v2 by @chrischdi 

For #514

https://github.com/openstack/neutron-lib/blob/56033ba643812a30577f6ab17648806c2ee494ba/neutron_lib/api/definitions/firewall_v2.py#L159
https://github.com/openstack/neutron-fwaas/blob/27906d0acff349c9e5ac7955d203fb2233c295db/neutron_fwaas/db/firewall/v2/firewall_db_v2.py#L148

- [x] acceptance test for updatePolicies, verify policy's rules length is 0 (from [discussion](https://github.com/gophercloud/gophercloud/pull/1789#discussion_r353527621))

<sub>Georg Schreiber georg.g.schreiber@daimler.com, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

### CODE REVIEW

**Comment by codemanufaktur** [comment]
here is the output of [acceptance/openstack/networking/v2/extensions/fwaas_v2/policy_test.go](https://gist.github.com/codemanufaktur/13698e5a2a13d0430defa2f3d8e91a27)

**Comment by codemanufaktur** [comment]
@Elethiomel @jtopjian ptal :-)

**Review by jtopjian — CHANGES_REQUESTED** [review]
@codemanufaktur Thank you for working on this and for the test output 🙂 

I've left a few comments - mostly just renaming fields. Please let me know if you have any questions.

File: openstack/networking/v2/extensions/fwaas_v2/rules/results.go [hunk 1]
```diff
@@ -7,21 +7,21 @@ import (
...
+	Protocol             string   `json:"protocol"`
+	Action               string   `json:"action"`
+	IPVersion            int      `json:"ip_version,omitempty"`
+	SourceIPAddress      string   `json:"source_ip_address,omitempty"`
+	DestinationIPAddress string   `json:"destination_ip_address,omitempty"`
+	SourcePort           string   `json:"source_port,omitempty"`
+	DestinationPort      string   `json:"destination_port,omitempty"`
+	Shared               bool     `json:"shared,omitempty"`
+	Enabled              bool     `json:"enabled,omitempty"`
+	FirewallPolicyID     []string `json:"firewall_policy_id"`
```
[the remark this answers is not in the document]
Comment (chrischdi) [reply openstack/networking/v2/extensions/fwaas_v2/rules/results.go:22]: Hi @jtopjian , if I did see it right, then `firewall_policy_id` was introduced as array:
See commit: https://github.com/openstack/neutron-fwaas/commit/55fed400b0511132b352b9a1fa0d2f55fc1bcbb8

Also here the reference to the neutron-fwaas codebase: https://github.com/openstack/neutron-fwaas/blob/stable/ussuri/neutron_fwaas/db/firewall/v2/firewall_db_v2.py#L282

File: openstack/networking/v2/extensions/fwaas_v2/policies/requests.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+func Delete(c *gophercloud.ServiceClient, id string) (r DeleteResult) {
+	_, r.Err = c.Delete(resourceURL(c, id), nil)
+	return
+}
+
+type InsertRuleOptsBuilder interface {
+	ToFirewallPolicyInsertRuleMap() (map[string]interface{}, error)
+}
+
+type InsertRuleOpts struct {
```
[the remark this answers is not in the document]
Comment (codemanufaktur) [reply openstack/networking/v2/extensions/fwaas_v2/policies/requests.go:145]: good point, thx! I changed it as you suggested :-)

**Review by jtopjian — APPROVED** [review]
LGTM - thank you!

**Pull request merged**